src/Composer/Command/UpdateCommand.php

```diff
@@ -248,7 +249,27 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             $install->disablePlugins();
         }
 
-        return $install->run();
+        $result = $install->run();
+
+        if ($result === 0) {
+            $bumpAfterUpdate = (bool) $input->getOption('bump-after-update') || (bool) $composer->getConfig()->get('bump-after-update');
+
+            if ($bumpAfterUpdate) {
+                $io->writeError('<info>Bumping dependencies</info>');
+                $bumpCommand = new BumpCommand();
+                $bumpCommand->setComposer($composer);
+                $result = $bumpCommand->doBump(
+                    $io,
+                    $input->getOption('bump-after-update') === 'dev'
+                        || $composer->getConfig()->get('bump-after-update') === 'dev',
+                    $input->getOption('bump-after-update') === 'no-dev'
+                        || $composer->getConfig()->get('bump-after-update') === 'no-dev',
+                    $input->getOption('dry-run'),
+                    $input->getArgument('packages')
+                );
+            }
```

Comment:
Doing `$input->getOption('bump-after-update') === 'dev' || $composer->getConfig()->get('bump-after-update') === 'dev',` here isn't very safe as you might end up with only dev AND only non-dev if you have the config set to no-dev and the option set to --bump-after-update=dev then both will be true.

```suggestion
            $bumpAfterUpdate = $input->getOption('bump-after-update');
            if (false === $bumpAfterUpdate) {
                $bumpAfterUpdate = $composer->getConfig()->get('bump-after-update');
            }

            if (false !== $bumpAfterUpdate) {
                $io->writeError('<info>Bumping dependencies</info>');
                $bumpCommand = new BumpCommand();
                $bumpCommand->setComposer($composer);
                $result = $bumpCommand->doBump(
                    $io,
                    $bumpAfterUpdate === 'dev',
                    $bumpAfterUpdate === 'no-dev',
                    $input->getOption('dry-run'),
                    $input->getArgument('packages')
                );
            }
```